src/mappings/trades.ts

```diff
@@ -19,8 +19,22 @@ export function getTradesByOrderId(orderId: string): Trade[] {
   return order.trades.map<Trade>((tradeId) => getTradeById(tradeId))
 }
 
+let _tradeFilterBatchId = BigInt.fromI32(-1)
 export function getTradeInBatchForOrderId(orderId: string, batchId: BigInt): Trade[] {
-  return getTradesByOrderId(orderId).filter((trade) => trade.tradeBatchId == batchId)
+  // NOTE: It appears that AssemblyScript does not yet fully implement closures.
+  // Even more so in the version being used by the Graph CLI. Specifically, the
+  // captured variable has no value (this can be verified by uncommenting out
+  // the log line to see that the captured `batchId` value is indeed `null`). To
+  // work around this issue, create a global variable for holding the batch ID
+  // being filtered for (🤮) and use that instead. Since Wasm is currently
+  // single threaded, this is not an issue yet. Unfortunately, there is not
+  // compiler error for capturing a variable, so its just something to watch out
+  // for.
+  _tradeFilterBatchId = batchId
+  return getTradesByOrderId(orderId).filter((trade) => {
+    //log.debug('batchId == {}', [batchId == null ? 'null' : batchId.toString()])
+    return trade.tradeBatchId == _tradeFilterBatchId
```

Comment:
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia2.giphy.com%2Fmedia%2F26FLbdnqSIk5ddfJm%2Fgiphy.gif&token=nasty" data-gitmeme-token="nasty"><img src="https://media2.giphy.com/media/26FLbdnqSIk5ddfJm/giphy.gif" title="Created by gitme.me with /nasty"/></a>

Reply:
O my!!! Nick, great investigation work!
And thanks for fixing my implementation ❤️